### PR TITLE
dockerize backend server

### PR DESCRIPTION
**Pull request by mueller-werner**
basic docker build for the backend server.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM nikolaik/python-nodejs:python3.9-nodejs15
+
+RUN wget -qO- https://github.com/strukturag/pdfdraw/archive/v0.1.2.tar.gz | tar -C /tmp -xvz --strip 1
```
Comment (fancycode) [inline docker/Dockerfile:3]: Could you please change this to use the files from the repository so the Docker container is not pinned to a certain version but runs the code checked out?

Maybe move the files from this PR into the `server` folder so you can use a workflow like this
```
git clone https://github.com/strukturag/pdfdraw.git
cd pdfdraw/server
docker-compose up -d
```

File: docker/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# run pdfdraw backend in a docker container using docker-compose.
```
Comment (fancycode) [inline docker/README.md:1]: If moved this could go into the existing `README.md` in the `server` folder.

**Review by fancycode — COMMENTED** [review]
Thanks, I left some comments in the code.

File: docker/Dockerfile (hunk 1 above)
Comment (mueller-werner) [reply docker/Dockerfile]: good point, done it exactly how you recommended

File: server/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+version: '2'
+services:
+  pdfdraw-server:
+    build:
+      context: .
+      dockerfile: docker/Dockerfile
+    environment:
```
Comment (fancycode) [inline server/docker-compose.yml:7]: Could you please rename the file to `docker-compose.yml.in` and add all possible options to the `environment` block (commented out) to see what can be customized without having to check the `docker/config.js.docker`.

That way people can customize their `docker-compose.yml` file without interfering with the file in the repository (similar to what is done with the `config.js.in`). Please also add an entry to the `.gitignore` file in the repository root for `docker-compose.yml`.

File: server/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+      port: 8080
+      secret: 'the-shared-secret'
+    #ports:
```
Comment (fancycode) [inline server/docker-compose.yml:10]: Trailing whitespace.

File: server/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    build:
+      context: .
+      dockerfile: docker/Dockerfile
+    environment:
+      port: 8080
+      secret: 'the-shared-secret'
+    #ports:  
+    #  - '8080:8080'
+
+#networks:
```
Comment (fancycode) [inline server/docker-compose.yml:13]: Trailing whitespace.

File: server/README.md [hunk 6]
```diff
@@ -63,3 +63,27 @@ fields:
 - `author` (string): Optional author name to add to the annotation metadata.
 - `modified` (number): Optional timestamp (seconds since the epoch, in UTC)
   when the annotation was last modified.
+
+# run pdfdraw server in a docker container using docker-compose.
```
Comment (fancycode) [inline server/README.md:67]:
```suggestion
# Run pdfdraw server in a docker container using docker-compose.
```

File: server/README.md [hunk 7]
```diff
@@ -63,3 +63,27 @@ fields:
...
+
+all settings are exposed into enviroment variables. That means you are able to config the backend server in [docker-compose.yml](docker-compose.yml) yaml file by editing the corresponding enviroment variables.
```
Comment (fancycode) [inline server/README.md:69]:
```suggestion
All settings are exposed into enviroment variables. That means you are able to config the backend server in [docker-compose.yml](docker-compose.yml) yaml file by editing the corresponding enviroment variables.
```

File: server/README.md [hunk 8]
```diff
@@ -63,3 +63,27 @@ fields:
...
+
+Based on your docker setup you still need to edit a few lines in the [docker-compose.yml](docker-compose.yml) file for networking.
```
Comment (fancycode) [inline server/README.md:71]: Mention copying the file from `.yml.in` to `.yml` as described below.

**Review by fancycode — CHANGES_REQUESTED** [review]
Almost there, tested and works nicely. A few minor changes are required until this can be merged. Thanks!

**Review by fancycode — APPROVED** [review]
Looks good now, thanks!

**Pull request merged**